mac

```diff
@@ -142,8 +142,8 @@ if brew_is_installed 'brew-cask'; then
   brew untap caskroom/versions
 fi
 
-fancy_echo "Updating Homebrew formulas ..."
-brew update
+fancy_echo "Updating Homebrew..."
+cd "$(brew --repo)" && git fetch && git reset --hard origin/master && brew update
```

Comment:
Yes, it's taken straight from the Homebrew repo: https://github.com/Homebrew/brew#update-bug